File: rotate_array.py

def rotate_array(array,k):
    #write code 
    if len(array)==0 or k==0 or k==len(array):
        return array

    k=k%len(array)
    
    temp=array[-k:]
    
    for i in reversed(range(0,len(array)-k)):
        array[i+k]=array[i]
    
    for i in range(len(temp)):
        array[i]=temp[i]
    
    return array
    

File: rotate_array.js
const rotateArray = function(array, k) {
    //write code 
    if ((array.length === 0) || k === 0 || k == array.length) {
        return array;
    }

    k = k % array.length;

    let temp = array.slice(-k);

    for (let i = array.length - k - 1; i >= 0; i--) {
        array[i + k] = array[i];
    }

    for (let i = 0; i < temp.length; i++) {
        array[i] = temp[i]
    }


    return array;
}